R/calcACOwnershipRates.R

```diff
@@ -0,0 +1,166 @@
+  ieaSingleCountry <- ieaData %>%
+    filter(!.data$region %in% regionmapIEA$regionIEA) %>%
+    select(-"variable") %>%
+    rename("valueIEASingle" = "value") %>%
+    mutate(region = as.character(.data$region))
+
+  ieaSingleCountry$region <- toolCountry2isocode(ieaSingleCountry$region)
+
+  #TODO: filter regional values and disaggregate them to fill missing values? #nolint
+
+
+  ### Odyssee ====
+
+  # filter "AC equipment rate" and take most recent data point per region where necessary
+  odysseeData <- odysseeData %>%
+    filter(.data$variable == "teqcli") %>%
+    select("region", "period", "value") %>%
+    left_join(regionmapEDGE, by = c("region" = "CountryCode")) %>%
+    group_by(.data$region) %>%
+    (\(x) {
+      # split data into (non-)single-country regions
+      nonSingleCountry <- filter(x, .data$region != .data$RegionCodeEUR_ETP)
+      singleCountry    <- filter(x, .data$region == .data$RegionCodeEUR_ETP)
+      # take only max period for non-single-country regions
+      nonSingleCountry <- nonSingleCountry %>%
+        slice_max(order_by = .data$period, n = 1, with_ties = FALSE)
+      bind_rows(nonSingleCountry, singleCountry)
+    })() %>%
+    ungroup() %>%
+    rename("valueOdyssee" = "value") %>%
+
+    # Montenegro (MNE) shows an ownership rate of 100% which seems unrealistic
+    filter(.data$region != "MNE")
+
+
+  ## Merge Datasets ====
+
+  # target : obtain merged data set with one data point per region
+  # merge priority : odyssee > iea > surveys
+
+  ownershipRates <- odysseeData %>%
+    full_join(ieaSingleCountry %>%
+                anti_join(odysseeData, by = "region"),
+              by = c("region", "period")) %>%
+    full_join(surveyData %>%
+                anti_join(odysseeData, by = "region") %>%
+                anti_join(ieaSingleCountry, by = "region"),
+              by = c("region", "period")) %>%
+    mutate(value = coalesce(.data$valueOdyssee, .data$valueIEASingle, .data$valueSurvey)) %>%
+    select("region", "period", "value")
+
+
+  # ensure all countries within EDGE region have the same period to avoid aggregation issues
+  ownershipRates <- ownershipRates %>%
+    left_join(regionmapEDGE, by = c("region" = "CountryCode")) %>%
+    group_by(across(all_of("RegionCodeEUR_ETP"))) %>%
+    mutate(period = ifelse(.data$region == .data$RegionCodeEUR_ETP,
+                           .data$period,
+                           as.integer(median(.data$period))),
+           variable = "ac ownership rate") %>%
+    ungroup() %>%
+    select(-"RegionCodeEUR_ETP") %>%
+    filter(!is.na(.data$value)) %>%
+    interpolate_missing_periods()
+
+
+  ## Aggregation Weights ====
+
+  pop <- pop %>%
+    select(-"variable") %>%
+    semi_join(ownershipRates, by = c("region", "period"))
+
+
+
+  # OUTPUT ---------------------------------------------------------------------
+
+  data <- ownershipRates %>%
+    as.quitte() %>%
+    as.magpie() %>%
+    toolCountryFill(0, verbosity = 2)
```

Comment:
Why do you fill with zero instead on NA? There should be a way to pass something like `na.rm = TRUE` to the aggregation to get the same result without setting value and weight to zero, right?